src/setting_sbp_cb.c

```diff
@@ -25,25 +25,45 @@
 #include <internal/setting_def.h>
 #include <internal/setting_sbp_cb.h>
 
+#define UPDATE_FILTER_NONE 0x0
+#define UPDATE_FILTER_READONLY 0x1
+#define UPDATE_FILTER_WATCHONLY 0x2
```

Comment:
possibly clearer with bit shifting to indicate usage as mask/flags
```suggestion
#define UPDATE_FILTER_READONLY (0x1 << 1)
#define UPDATE_FILTER_WATCHONLY (0x1 << 2)

#define update_filter_check(filter_mask, filter_enum)  (0 == (filter_mask & filter_enum))
```
encapsulate checking either with macro or fully fledged helper func